SpatialGDK/Source/SpatialGDK/Public/Interop/RPCs/RPC_RingBufferWithACK_Receiver.h

```diff
@@ -102,15 +102,15 @@ class MonotonicRingBufferWithACKReceiver : public TRPCBufferReceiver<PayloadType
 		for (auto Iterator = this->ReceivedRPCs.CreateIterator(); Iterator; ++Iterator)
 		{
 			Worker_EntityId EntityId = Iterator->Key;
-			if (!CanExtract(EntityId))
-			{
-				continue;
-			}
 			ReceiverState& State = ReceiverStates.FindChecked(EntityId);
 			auto& Payloads = Iterator->Value;
 			uint32 ProcessedRPCs = 0;
 			for (const auto& Payload : Payloads)
 			{
+				if (!CanExtract(EntityId))
+				{
+					continue;
+				}
 				const PayloadType& PayloadData = Payload.GetData();
 				if (Process(EntityId, PayloadData, Payload.GetAdditionalData()))
```

Comment:
So to be clear this bit crashes right @AndreiTeodorLazar?

Reply:
And basically the CanExtract function which is actually in our case a call to CanExtractRPCOnServer is checking that the ActorReceivingRPC is not null.

Reply:
Right but `Process` is a function pointer pointing to `ApplyRPC` when it crashes correct?

Reply:
Yes that is correct